cpp/cmake_modules/FindNumPy.cmake

```diff
@@ -96,5 +96,11 @@ find_package_message(NUMPY
 set(NUMPY_FOUND TRUE)
 
 add_library(Python3::NumPy INTERFACE IMPORTED)
-target_include_directories(Python3::NumPy INTERFACE ${NUMPY_INCLUDE_DIRS})
+if(CMAKE_VERSION VERSION_LESS 3.11)
+    target_include_directories(Python3::NumPy PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
+        ${NUMPY_INCLUDE_DIRS})
+    set_target_properties(Python3::Module PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
```

Comment:
Should this be:
```suggestion
    set_target_properties(Python3::Module PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
        ${NUMPY_INCLUDE_DIRS})
```

Reply:
Oh, this is broken code...
We should use `set_target_properties()` + `Python3::NumPy` here.